### ISSUE TITLE
vim.lsp.protocol.CompletionTag is nil

### ISSUE THREAD

**Issue by sieuwerts** [Saghen/blink.compat#16]
### Make sure you have done the following

- [X] I have updated to the latest version of `blink.cmp` and `blink.compat`
- [X] I have read the README

### Bug Description

This line is throwing an error for me https://github.com/Saghen/blink.compat/blob/main/lua/cmp/types/lsp.lua#L20

When inspecting `vim.lsp.protocol` it does not contain `CompletionTag`, hence when referenced above, it's nil and the error occurs. Granted, I try to add [copilot-cmp](https://github.com/zbirenbaum/copilot-cmp) as a provider, (without success), but I don't think that this error is related to that specific plugin.

```
09:55:56 msg_show Error executing lua callback: ...re/nvim/lazy/blink.compat/lua/blink/compat/lib/utils.lua:9: bad argument #1 to 'pairs' (table expected, got nil)
stack traceback:
	[C]: in function 'pairs'
	...re/nvim/lazy/blink.compat/lua/blink/compat/lib/utils.lua:9: in function 'shallow_copy'
	...local/share/nvim/lazy/blink.compat/lua/cmp/types/lsp.lua:20: in main chunk
	[C]: in function 'require'
	...com/.local/share/nvim/lazy/blink.compat/lua/cmp/init.lua:9: in main chunk
```

Perhaps this is a "me" issue, and not a blink.compat issue, but I'm not really sure on how to tell.

### Relevant configuration

```lua
sources = {
				providers = {
					copilot = {
						name = "copilot",
						module = "blink.compat.source",
						score_offset = -3,
						opts = {},
					},
				},

				completion = {
					enabled_providers = { "lsp", "path", "snippets", "buffer", "copilot" },
				},
			},
```

### neovim version

NVIM v0.10.2 Build type: Release LuaJIT 2.1.1732813678

### `blink.cmp` version: branch, tag, or commit

v0.*

### `blink.compat` version: branch, tag, or commit

main

**Comment by reedts** [comment]
I encountered the same issue but with the `luasnip` provider:

```lua
luasnip = {
	name = 'luasnip',
	module = 'blink.compat.source',
                                    
	score_offset = -3,
                                    
	opts = {
		use_show_condition = false,
		show_autosnippets = true,
	},
},
```
### neovim version
NVIM v0.10.2 Build type: Release LuaJIT 2.1.1731601260

### `blink.cmp` version: branch, tag, or commit
"v0.*"

### `blink.compat` version: branch, tag, or commit
"*"

**Comment by AbysmalBiscuit** [comment]
I also have the same problem with `luasnip`.

Downgrading to [v2.1.0](https://github.com/Saghen/blink.compat/releases/tag/v2.1.0) fixed the error:

```lua
{ 'saghen/blink.compat', version = 'v2.1.0' },
```

Then running: `:Lazy update`

**Comment by DanWlker** [comment]
Same problem for luasnip, downgrading to this commit fixed the issue
6acb0fa5c1563958f97c46e4e637350485b20401

should be caused by this commit:
0fcc94eb9aa899e7453ea00f6f85ff6b66d8c0e9

**Comment by stefanboca** [comment]
Should be fixed in the v2.1.2. I believe `CompletionTag` was only added in neovim>=0.11, which is what caused the issue.
